repo/blob/s3/s3_storage_test.go

```diff
@@ -62,39 +90,52 @@ func endpointReachable() bool {
 }
 
 func TestS3Storage(t *testing.T) {
-	testStorage(t, accessKeyID, secretAccessKey, "")
+	// run tests on play.minio.io if AWS creds no provided
```

Comment:
We should preserve the upstream minio test with no modifications. Perhaps rename it to `TestMinioS3Storage`
The goal is to be able to run this test without external (`Makefile`) modifications.

And
* add an S3 test that is dependent on whether or not creds are specified.
* modify the session token test (or add a new test) to use env vars as you have it below. Ideally, it'd be good to still keep the STS test with minio, with minor modifications as discussed in the upstream issue.

Reply:
Sounds good. Only concern I have is with minio sts test. It can fail randomly sometimes.

Reply:
Left some comment sin the upstream issue. I'll review this PR shortly.

Reply:
Addressed your comments from the upstream issue